src/uri/uri.cc

```diff
@@ -249,6 +249,22 @@ auto URI::path() const -> std::optional<std::string> {
   return path_;
 }
 
+auto URI::path(std::string_view path) -> URI & {
+  if (path.empty()) {
+    this->path_ = std::nullopt;
+    return *this;
+  }
+
+  const auto parsed_path = URI{std::string{path}};
+  const bool has_leading_slash = path.front() == '/';
+  const std::string path_to_set = parsed_path.path().value();
+
+  this->path_ = has_leading_slash ? path_to_set.substr(1) : path_to_set;
+  return *this;
+}
+
+auto URI::path(std::string &&path) -> URI & { return this->path(path); }
```

Comment:
I don't think you are actually implementing move semantics. You are calling back to the copy overload, thus forcing a copy. You will need the implementation here to be more elaborate and likely NOT call the other one. 

I think you will end up repeating some of the boilerplate, but passing the string ownership to the `URI` constructor, and then moving `path` from your temporary URI to the path member in the current class

Reply:
> You will need the implementation here to be more elaborate and likely NOT call the other one.

Funny because I started with it haha :) 

> I think you will end up repeating some of the boilerplate, but passing the string ownership to the URI constructor, and then moving path from your temporary URI to the path member in the current class

Yeah it makes whole sense.

Move semantics confused me here as I was assuming that it was the string passed as a parameter that was "moved" (`std::move(xxxx)`) and that on our side we don't have to do anything. My reasoning was wrong! 

Note: I realised now that the move constructor I implemented above could have help me to understand it!

Reply:
I don't remember if I recommended it before, but http://cppmove.com is quite fun to read on this topic